tests/scripts/all.sh

```diff
@@ -580,6 +581,10 @@ msg "test/build: key-exchanges (gcc)" # ~ 1 min
 cleanup
 record_status tests/scripts/key-exchanges.pl
 
+msg "build: Unix make, gcc and g++ test" # ~ 30s
+cleanup
+make TEST_CPP=1
+
```

Comment:
I would like to add a check that `cpp_dummy_build.cpp` lists all the headers except for whitelisted ones.
```
check_headers_in_cpp () {
    ls include/mbedtls >headers.txt
    <programs/test/cpp_dummy_build.cpp sed -n 's/"$//; s!^#include "mbedtls/!!p' |
    sort |
    diff headers.txt -
    rm headers.txt
}
record_status check_headers_in_cpp
```

Reply:
Will there be any whitelisted ones? Currently, this fails only because check-config.h is not included there (there's no need to test it since it does not have any extern directives), but there's no problem in adding it there.

Reply:
`check_config.h` is the only whitelisted one.

Reply:
Added a check, included the check-config.h in the dummy build file.